### ISSUE TITLE
Kapitan 0.29.3 not working with Python 3.9.5

### ISSUE THREAD

**Issue by xccatana** [kapicorp/kapitan#737]
**Describe the bug/feature**
Kapitan 0.29.3 doesn't work with Python 3.9.5. but it works with Python 3.8.5

**To Reproduce**
1. Create virtualenv and source to it
2. pip install kapitan==0.29.3
3. kapitan compile
```
(kapitan-dev) ccatana monitoring (master) >> kapitan compile
Unknown (Non-Kapitan) Error occurred
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/targets.py", line 453, in compile_target
    input_compiler.compile_obj(comp_obj, ext_vars, **kwargs)
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/inputs/base.py", line 54, in compile_obj
    self.compile_input_path(input_path, comp_obj, ext_vars, **kwargs)
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/inputs/base.py", line 69, in compile_input_path
    self.compile_file(
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/inputs/jinja2.py", line 35, in compile_file
    context["inventory"] = inventory(self.search_paths, target_name)
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/resources.py", line 260, in inventory
    inventory_path = inventory_path_arg.inventory_path
AttributeError: 'NoneType' object has no attribute 'inventory_path'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/targets.py", line 126, in compile_targets
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/targets.py", line 126, in <listcomp>
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 870, in next
    raise value
AttributeError: 'NoneType' object has no attribute 'inventory_path'

'NoneType' object has no attribute 'inventory_path'

OR

(kapitan-dev) ccatana monitoring (master) >> kapitan compile

Jsonnet error: failed to compile /Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:
 RUNTIME ERROR: 'NoneType' object has no attribute 'inventory_path'
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/kapitan-dev/lib/python3.9/site-packages/kapitan/lib/kapitan.libjsonnet:9:59-100	function <anonymous>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:5:16-31	thunk <params>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:8:86-92	thunk <o>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/lib/common.libjsonnet:5:22	thunk <o>
	std.jsonnet:1299:21	
	std.jsonnet:1299:5-33	function <anonymous>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/lib/common.libjsonnet:5:8-27	function <anonymous>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:8:76-123	thunk <o>
	std.jsonnet:1293:24	
	std.jsonnet:1293:5-33	function <anonymous>
	...
	std.jsonnet:790:15-18	thunk <arr>
	std.jsonnet:786:28-31	
	std.jsonnet:786:17-32	function <aux>
	std.jsonnet:790:5-28	function <anonymous>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:(9:25)-(12:6)	thunk <apps_by_service>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:18:36-51	thunk <o>
	std.jsonnet:1293:24	
	std.jsonnet:1293:5-33	function <anonymous>
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:18:19-52	
	/Users/ccatana/Documents/workspace/git_corp_com/monitoring/components/install/asserter.jsonnet:18:8-53
```

**Expected behavior**
From what I see inside repository, it is specified that Kapitan works with Python 3.6, and that's the only version that I saw mentioned inside the repository. Do you guys support other Python version as well?
I was using Python 3.9.5 and I couldn't make Kapitan work with this version, but it works just fine with Python 3.8.5.

Environment:
 - `python --version`: Python 3.9.5
 - `pip3 --version`: pip 21.1.2 
 - `Are you using pyenv or virtualenv?`: virtualenv 20.4.7 
 - `Distribution`: MacOS Catalina 10.15.7

Thank you!

**Comment by ramaro** [comment]
Hey @xccatana I believe this is related to a multiprocessing library change within Python that behaves differently in 3.9.x. However, if you run Kapitan 0.29.5 this shouldn't be an issue - can you confirm? Thanks for reporting :)

**Comment by sebradloff** [comment]
This appears to still be an issue, while exec into the kapitan docker image v0.29.5

```
 docker run -it --rm -v /Users/sradloff/Desktop/code/k8s-kapitan-fluxcd-example:/src --entrypoint sh deepmind/kapitan:v0.29.5
$ kapitan c
Unknown (Non-Kapitan) Error occurred
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 465, in compile_target
    input_compiler.compile_obj(comp_obj, ext_vars, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/base.py", line 54, in compile_obj
    self.compile_input_path(input_path, comp_obj, ext_vars, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/base.py", line 76, in compile_input_path
    **kwargs,
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/jinja2.py", line 35, in compile_file
    context["inventory"] = inventory(self.search_paths, target_name)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/resources.py", line 260, in inventory
    inventory_path = inventory_path_arg.inventory_path
AttributeError: 'NoneType' object has no attribute 'inventory_path'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 136, in compile_targets
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 136, in <listcomp>
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/usr/local/lib/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
AttributeError: 'NoneType' object has no attribute 'inventory_path'


'NoneType' object has no attribute 'inventory_path'
```

**Comment by ramaro** [comment]
closing as fixed via #837

**Comment by xccatana** [comment]
Thank you for fix @ramaro!
